runtime/src/main/java/org/corfudb/runtime/view/LayoutBuilder.java

```diff
@@ -323,36 +335,52 @@ public LayoutBuilder removeSequencerServers(Set<String> endpoints)
         return this;
     }
 
+    /**
+     * Remove an endpoint from a segment.
+     *
+     * @param endpoint The endpoint to remove
+     * @param layoutStripe the stripe to remove the endpoint from
+     * @param mode mode of this segment
+     * @throws LayoutModificationException
+     */
+    public void removeFromStripe(String endpoint, LayoutStripe layoutStripe,
+                                 Layout.ReplicationMode mode) throws LayoutModificationException {
+        if (layoutStripe.getLogServers().remove(endpoint)) {
+            if (layoutStripe.getLogServers().isEmpty()) {
+                throw new LayoutModificationException(
+                        "Attempting to remove all logunit in stripe. "
+                                + "No replicas available.");
+            }
+            if (mode == Layout.ReplicationMode.CHAIN_REPLICATION
+                    && layoutStripe.getLogServers().size() < CHAIN_REPLICATION_MIN_NUM_NODES) {
+                throw new LayoutModificationException(
+                        "Change will cause redundancy loss!");
+            } else if (mode == Layout.ReplicationMode.QUORUM_REPLICATION) {
+                throw new IllegalStateException("Unsupported operation!");
+            }
+        }
+    }
+
     /**
      * Removes the logunit endpoint from the layout.
      *
      * @param endpoint Log unit server to be removed
-     * @return Workflow manager
+     * @return Builder
      * @throws LayoutModificationException Cannot remove non-replicated logunit
      */
     public LayoutBuilder removeLogunitServer(String endpoint)
-            throws LayoutModificationException {
+            throws LayoutModificationException , CloneNotSupportedException{
```

Comment:
throw UnrecoverableCorfuError (and you don't have to put this in the throws clause)

Reply:
public LayoutBuilder removeLogunitServer(String endpoint) throws UnrecoverableCorfuError ?
That's a compiler error, because both extent exception directly.

Reply:
No, if you wrap whatever is throwing clonenotsupportedexception in UnrecoverableCorfuError, you won't have to declare the exception in the throws list

Reply:
Done